### DISCUSSION TITLE
Error	CS1061	'WorkflowExecutionContext' does not contain a definition for 'Output' and no accessible extension method 'Output' accepting a first argument of type 'WorkflowExecutionContext' could be found (are you missing a using directive or an assembly reference?)

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5258] [Design & tech]
public class MyActivity : Activity
{
    protected override async ValueTask<IActivityExecutionResult> OnExecuteAsync(ActivityExecutionContext context)
    {
        // Perform the activity's work here and determine the output
        var myOutput = new MyOutputType(); // Replace with actual output type and value

        // Set the output
        context.WorkflowExecutionContext.Output["MyActivityOutput"] = myOutput;

        return Done();
    }
}

getting below error while setting output
   context.WorkflowExecutionContext.Output["MyActivityOutput"] = myOutput;

Error	CS1061	'WorkflowExecutionContext' does not contain a definition for 'Output' and no accessible extension method 'Output' accepting a first argument of type 'WorkflowExecutionContext' could be found (are you missing a using directive or an assembly reference?)

which nuget package is required?

**Comment by glime-ai** [comment]
Based on the context provided, it seems like you are trying to set the output of a custom activity in Elsa Workflow. The error you are encountering indicates that the `WorkflowExecutionContext` does not have an `Output` property that you can directly assign values to.

In Elsa 2.x and above, to set the output of an activity, you should use the `SetOutput` method provided by the `ActivityExecutionContext` rather than trying to set it directly on the `WorkflowExecutionContext`. Here's how you can modify your code:

```csharp
protected override async ValueTask<IActivityExecutionResult> OnExecuteAsync(ActivityExecutionContext context)
{
    // Perform the activity's work here and determine the output
    var myOutput = new MyOutputType(); // Replace with actual output type and value

    // Set the output
    context.SetOutput(myOutput);

    return Done();
}
```

Make sure that you have the appropriate using directives at the top of your file to include the necessary namespaces for Elsa Workflow. Typically, you would need something like:

```csharp
using Elsa.Services;
using Elsa.Services.Models;
```

If you are still encountering issues, ensure that you have the latest stable version of the Elsa packages installed via NuGet. You can check for the latest versions on the official NuGet website or by using the NuGet package manager in Visual Studio.

If you are using a version of Elsa prior to 2.x, the way to handle outputs may be different, and you should refer to the documentation or upgrade to a newer version.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by NiveditaGalwade** [reply]
We are not getting SetOutput method to set output .
I am using Elsa 3.1.2 version which is latest version
  context.SetOutput(myOutput);
